packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -148,7 +166,21 @@ export class ChainService implements ChainServiceInterface {
       to: assetHolderAddress,
       value: isEthFunding ? arg.amount : undefined,
     };
-    return this.sendTransaction(depositRequest);
+
+    const tx = await this.sendTransaction(depositRequest);
+
+    if (!tx) throw new Error('fundChannel failed to submit deposit transaction');
```

Comment:
It seems wrong to use `undefined` return values to propagate errors. `sendTransaction` throws an error. Is it possible to catch the error and throw a new error here?

Also, by discarding the original error, we are likely losing valuable debugging information.

Reply:
Yeah, I agree, changing

Reply:
Can this condition ever get triggered?